dataland-frontend/tests/e2e/utils/EuTaxonomyFinancialsUpload.ts

```diff
@@ -27,27 +33,101 @@ export function submitEuTaxonomyFinancialsUploadForm(): Cypress.Chainable {
+  cy.get('input[name="currency"]').type("www");
+  cy.get('button[data-test="uploaded-files-remove"]').click();
+  cy.get('div[data-test="uploaded-files"]').should("not.exist");
 
-  cy.get("select[name=financialServicesTypes]").select(data.financialServicesTypes || []);
+  cy.get('[data-test="fiscalYearEnd"] button').should("have.class", "p-datepicker-trigger").click();
+  cy.get("div.p-datepicker").find('button[aria-label="Next Month"]').click();
+  cy.get("div.p-datepicker").find('span:contains("12")').click();
+  cy.get('input[name="fiscalYearEnd"]').invoke("val").should("contain", "12");
+  cy.get('input[name="reportDate"]').should("not.exist");
+  cy.get('input[name="reference"]').should("not.exist");
+  cy.get('input[name="fiscalYearEnd"]').should("not.be.visible");
 
-  if (data.assurance?.assurance !== undefined) {
-    cy.get("select[name=assurance]").select(data.assurance.assurance.toString());
-  }
+  cy.get('[data-test="MultiSelectfinancialServicesTypes"]')
+    .click()
+    .get("div.p-multiselect-panel")
+    .find("li.p-multiselect-item")
+    .first()
+    .click();
+  cy.get('[data-test="addKpisButton"]').click({ force: true });
+  cy.get('[data-test="removeSectionButton"]').click({ force: true });
 
+  cy.get('[data-test="MultiSelectfinancialServicesTypes"]')
+    .click()
+    .get("div.p-multiselect-panel")
+    .find("li.p-multiselect-item")
+    .each(($el) => {
+      cy.wrap($el).click({ force: true });
+    });
+  cy.get('[data-test="addKpisButton"]').click({ force: true });
+  cy.get('[data-test="dataPointToggle"]')
+    .eq(1)
+    .should("exist")
+    .should("contain.text", "Data point is available")
+    .find('[data-test="dataPointToggleButton"]')
+    .click();
+  cy.get('[data-test="dataPointToggle"]')
+    .eq(1)
+    .find('[data-test="dataPointToggleTitle"]')
+    .should("contain.text", "Data point is not available");
+  cy.get('[data-test="dataPointToggle"]').eq(1).find('[data-test="dataPointToggleButton"]').click();
+  cy.get('[data-test="dataPointToggle"]')
+    .eq(1)
+    .find('[data-test="dataPointToggleTitle"]')
+    .should("contain.text", "Data point is available");
   if (data.reportingObligation !== undefined) {
     cy.get(`input[name="reportingObligation"][value=${data.reportingObligation.toString()}]`).check();
   }
+  cy.get(
+    `input[name="fiscalYearDeviation"][value=${
+      data.fiscalYearDeviation ? data.fiscalYearDeviation.toString() : "Deviation"
+    }]`
+  ).check();
+  cy.get('input[name="scopeOfEntities"][value="No"]').check();
+  cy.get('div[id="jumpLinks"] li:last a').click();
+  cy.window().then((win) => {
+    const scrollPosition = win.scrollY;
+    expect(scrollPosition).to.be.greaterThan(0);
+  });
+  cy.get('button[data-test="submitButton"]').click();
+  cy.window().then((win) => {
+    const scrollPosition = win.scrollY;
+    expect(scrollPosition).to.be.greaterThan(0);
+  });
+  cy.get(
```

Comment:
All the stuff that is in this area of the test could simply happen in a component test, because it's just frontend clicking with no actual requests.

We could externalize it to a component test, and leave the `fillEuTaxonomyForFinancialsUploadForm` to simply fill the form (as it was before).

That way all the pure frontend/convenience/design features (such as data toggles, date picker, add kpi section functionality etc.) would be tested in a component test and would not slow down the e2e test pipeline,
while the form would actually still be used for one upload in the e2e tests.